rholang/src/main/scala/coop/rchain/rholang/interpreter/util/AddressTools.scala

```diff
@@ -25,12 +26,16 @@ class AddressTools(prefix: Array[Byte], keyLength: Int, checksumLength: Int) {
     */
   def fromPublicKey(pk: PublicKey): Option[Address] =
     if (keyLength == pk.bytes.length) {
-      val keyHash = Blake2b256.hash(pk.bytes)
-      val payload = prefix ++ keyHash
-
-      Some(Address(prefix, keyHash, computeChecksum(payload)))
+      val ethAddress = Base16.encode(Keccak256.hash(pk.bytes.drop(1))).takeRight(40)
+      Some(fromEthAddress(ethAddress))
     } else None
 
+  def fromEthAddress(ethAddress: String): Address = {
+    val keyHash = Keccak256.hash(Base16.unsafeDecode(ethAddress))
+    val payload = prefix ++ keyHash
+    Address(prefix, keyHash, computeChecksum(payload))
```

Comment:
There was a discussion with @cboscolo about the address algorithm and he said he will update the spec wiki page but I don't see any change: https://rchain.atlassian.net/wiki/spaces/CORE/pages/652640311/Wallet+proposal+Note+we+decided+to+rename+terms+such+that+wallet+is+not+used+for+on-chain+components

So, are we keeping the algorithm the same or will we change it? 

The downside of the current algorithm is the readability of the resulting address. My preference is to have a human readable ascii prefix like `revxx` instead of the current one.

If we decide to change, we should do it in a future Jira/PR, of course, but I'm just pointing at the issue